libs/functions/calculate_projections.py

```diff
@@ -108,46 +119,57 @@ def get_state_projections_df(input_dir, initial_intervention_type, state_interve
     """
     for each state in our data look at the results we generated via run.py
     to create the projections
+
+    columns=CALCULATED_PROJECTION_HEADERS_STATES
     """
 
-    # save results in a list of lists, converted to df later
-    results = []
-    for state in list(US_STATE_ABBREV.values()):
-        intervention_type = _get_intervention_type(initial_intervention_type, state, state_interventions_df)
+    sdf = pd.DataFrame(US_STATE_ABBREV.values(), columns=['state'])
+    sdf.loc[:,'intervention_type'] = sdf.state.parallel_apply(
+        lambda x: _get_intervention_type(initial_intervention_type, x, state_interventions_df)
+        )
+    sdf.loc[:, 'path'] = sdf.parallel_apply(
+        lambda x: get_file_path(input_dir, x.state, x.intervention_type, fips=None),
+        axis=1).values
+    ndf = sdf.parallel_apply(
+        lambda x:_calculate_projection_data(x.state, x.path, fips=None), axis=1)
+
+    num_processed_states = ndf.notnull().sum()['State']
+
+    if (num_processed_states < 51):
+        raise Exception(f"Missing too states! {num_processed_states} states were in input_dir: {input_dir}")
+    return ndf
+
+
+def get_file_path(input_dir, state, intervention_type, fips=None):
+    if fips:
+        file_name = f"{state}.{fips}.{intervention_type}.json"
+    else:
         file_name = f"{state}.{intervention_type}.json"
-        path = os.path.join(input_dir, file_name)
-        # if the file exists in that directory then process
-        projection_data = _calculate_projection_data(path)
-        if projection_data:
-            results.append([state] + projection_data)
-    return pd.DataFrame(results, columns=CALCULATED_PROJECTION_HEADERS_STATES)
+    return os.path.join(input_dir, file_name)
+
 
 def get_county_projections_df(input_dir, initial_intervention_type, state_interventions_df):
     """
     for each state in our data look at the results we generated via run.py
     to create the projections
+
+    #columns=CALCULATED_PROJECTION_HEADERS_COUNTIES)
     """
     fips_pd = FIPSPopulation.local().data  # to get the state, county & fips
 
-    # save results in a list of lists, converted to df later
-    results = []
+    fdf = fips_pd[['state' ,'fips']]
+    fdf.loc[:,'intervention_type'] = fdf.state.parallel_apply(
+        lambda x: _get_intervention_type(initial_intervention_type, x, state_interventions_df)
+        )
+    fdf.loc[:, 'path'] = fdf.parallel_apply(
```

Comment:
This also probably doesn't need to be parallel, there's no intensive operation there